### ISSUE TITLE
Smartctl error code 4 

### ISSUE THREAD

**Issue by Al3cr1s** [WEEE-Open/pesto#120]
Pinolo dies but the backend is still up.

**Comment by il-palmi** [comment]
Can you give more details about the error?

**Comment by Al3cr1s** [comment]
I've been told that if the disk works but has issues, it returns an exit code 4, which is probably not accounted for in Pinolo, causing it to crash.

**Comment by il-palmi** [comment]
Are you sure that smartctl returns exit code 4?

**Comment by lvps** [comment]
FYI, smartctl return code is a bitmask, 4 = 100 in binary, so according to the manual:

> Bit 3:
> SMART status check returned "DISK FAILING". 

The status is already checked in the smartctl parser (hopefully), no need to check the return code, just to avoid crashing

**Comment by il-palmi** [comment]
I found the error, apparently there is such a thing as MAX_LENGTH in Twisted's LineOnlyReceiver. If the incoming message is longer than the default length (16384 bytes), the connection is dropped cleanly (I don't know why they decided to handle this situation like that...). For now it is a client-side problem, I don't think a similar situation can occur for the server. I've committed the fix in my local repo, I will push it soon.

https://docs.twisted.org/en/stable/api/twisted.protocols.basic.LineOnlyReceiver.html
